src/rust/engine/process_execution/src/docker.rs

```diff
@@ -185,13 +196,34 @@ async fn pull_image(docker: &Docker, image: &str, policy: ImagePullPolicy) -> Re
     let mut result_stream = docker.create_image(Some(create_image_options), None, None);
     while let Some(msg) = result_stream.next().await {
       log::trace!("pull {}: {:?}", image, msg);
-      if let Err(err) = msg {
-        return Err(format!(
-          "Failed to pull Docker image `{}`: {:?}",
-          image, err
-        ));
-      }
+      match msg {
+        Ok(msg) => match msg {
+          CreateImageInfo {
+            error: Some(error), ..
+          } => {
+            let error_msg = format!("Failed to pull Docker image `{image}`: {error}");
+            log::error!("{error_msg}");
+            return Err(error_msg);
+          }
+          CreateImageInfo {
+            status: Some(status),
+            ..
+          } => {
+            log::info!("Docker pull status: {status}");
+          }
+          // Ignore content in other event fields, namely `id`, `progress`, and `progress_detail`.
+          _ => (),
+        },
+        Err(err) => {
+          return Err(format!(
+            "Failed to pull Docker image `{}`: {:?}",
+            image, err
+          ))
+        }
+      };
     }
+
+    log::info!("Finished pull of Docker image `{image}`.");
```

Comment:
Consider using a workunit, rather than a pair of log messages?

Reply:
Good idea. How should the status message be logged then? debug level?

Reply:
Also the actual `pull_image` function is debounced via a `OnceCell`. Will that affect associating the work unit with the correct parent?

Reply:
> Good idea. How should the status message be logged then? debug level?

The workunit's level (and whether the UI is in use) drives how it is logged. When the UI is enabled, DEBUG or greater are rendered in the UI, but only INFO or greater have their completion logged. When the UI is disabled, debug won't render at all (unless it is a "Long running workunit..."), but INFO will render both start and end.

> Also the actual `pull_image` function is debounced via a `OnceCell`. Will that affect associating the work unit with the correct parent?

It won't, no.

Reply:
Should the in_workunit! call be inside the OnceCell get_or_try_init or outside of it?

Reply:
Inside. Otherwise the workunit will be created even if nothing needs to be done.